presto-native-execution/presto_cpp/main/operators/tests/UnsafeRowShuffleTest.cpp

```diff
@@ -479,12 +486,108 @@ class UnsafeRowShuffleTest : public exec::test::OperatorTestBase {
                                .runtimeStats;
       ASSERT_EQ(1, exchangeStats.count(fmt::format("{}.read", shuffleName)));
 
-      for (auto& resultVector : results) {
-        auto vector = copyResultVector(resultVector);
+      for (auto partitionId = 0; partitionId < results.size(); ++partitionId) {
+        auto vector = copyResultVector(results.at(partitionId));
         outputVectors.push_back(vector);
+        auto partitionVector = vector->childAt(0)->asFlatVector<int32_t>();
+        if (partitionVector->size() == 0) {
+          emptyPartition.insert(partitionId);
+        }
+        for (auto idx = 0; idx < partitionVector->size(); ++idx) {
+          if (partitionVector->isNullAt(idx)) {
+            nullElementCount[partitionId] = nullElementCount[partitionId] + 1;
+          }
+        }
+      }
+    }
+
+    if (replicateNullsAndAny && numPartitions > 1) {
+      EXPECT_EQ(emptyPartition.size(), 0);
+      EXPECT_THAT(
+          nullElementCount,
+          ElementsAre(Pair(_, nullElementCount.begin()->second)));
+    } else {
```

Comment:
Consider to add assertContainResults in addition to the empty and null check above? The outputVectors is a superset of expectedContainedVectors? You could leave this for a followup. Thanks!

Reply:
Added a todo